src/components/TodoContent.tsx

```diff
@@ -0,0 +1,185 @@
+  ] = useState(false);
+  const [areTodosToggling, setAreTodosToggling] = useState(false);
+
+  const isContentRendered = todos.length > 0;
+
+  function setErrorsToFalseExceptLoadingError() {
+    setIsEmptyTitleErrorShown(false);
+    setIsRemoveErrorShown(false);
+    setIsTogglingErrorShown(false);
+    setIsAddingErrorShown(false);
+
+    setHasLoadingError(true);
+  }
+
+  function setErrorsToFalseExceptEmptyTitleError() {
+    setHasLoadingError(false);
+    setIsTogglingErrorShown(false);
+    setIsRemoveErrorShown(false);
+    setIsAddingErrorShown(false);
+
+    setIsEmptyTitleErrorShown(true);
+  }
+
+  function setErrorsToFalseExceptAddingError() {
+    setHasLoadingError(false);
+    setIsTogglingErrorShown(false);
+    setIsRemoveErrorShown(false);
+    setIsEmptyTitleErrorShown(false);
+
+    setIsAddingErrorShown(true);
+  }
+
+  useEffect(() => {
+    if (!user) {
+      return;
+    }
+
+    if (newTodoField.current && hasLoadingError) {
+      newTodoField.current.focus();
+    }
+
+    getTodos(user.id)
+      .then((fetchedTodos) => {
+        setVisibleTodos(fetchedTodos);
+      })
+      .catch(() => {
+        setErrorsToFalseExceptLoadingError();
+      });
+  }, []);
+
+  useEffect(() => {
+    if (!user) {
+      return;
+    }
+
+    getTodos(user.id).then(fetchedTodos => setTodos(fetchedTodos));
+  }, [visibleTodos]);
+
+  function onSubmitHandler(event: FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    setClickedIndex(visibleTodos.length);
+    const inputValue = newTodoField?.current?.value;
+
+    if (inputValue?.length === 0) {
+      setErrorsToFalseExceptEmptyTitleError();
+
+      return;
+    }
+
+    setIsNewTodoLoaded(false);
+
+    const newTodoObj = {
+      title: newTodoField.current?.value || '',
+      userId: user?.id || 0,
+      completed: false,
+      id: 0,
+    };
+
+    setVisibleTodos(prev => [...prev, newTodoObj]);
+    setTodos(visibleTodos);
+
+    if (user && newTodoField.current) {
+      postTodo(user.id, newTodoObj)
+        .then(data => {
+          setIsAddingErrorShown(false);
+
+          setVisibleTodos((prev: Todo[]) => {
+            setIsNewTodoLoaded(true);
+            const prevArr = prev.slice(0, -1);
+
+            return [...prevArr, data];
+          });
+        })
+        .catch(() => {
+          setErrorsToFalseExceptAddingError();
+
+          setVisibleTodos(prev => prev.slice(0, prev.length - 1));
+          setTodos(visibleTodos);
+        });
+
+      newTodoField.current.value = '';
+    }
+  }
+
+  return (
+    <div className="todoapp__content">
+      <header className="todoapp__header">
+        {visibleTodos.length > 0 && (
+          <ToggleAllButton
+            visibleTodos={visibleTodos}
+            setVisibleTodos={setVisibleTodos}
+            setAreTodosToggling={setAreTodosToggling}
+          />
+        )}
+
+        <form
+          onSubmit={(event) => onSubmitHandler(event)}
```

Comment:
Please fix all cases if present
```suggestion
          onSubmit={onSubmitHandler}
```